In [ ]:
!pip install vega altair vega_datasets
!pip install pandas-profiling[notebook]


     |████████████████████████████████| 1.3 MB 4.7 MB/s 


In [ ]:
import pandas as pd
import numpy as np
import altair as alt
alt.renderers.enable('default')



RendererRegistry.enable('default')

In [ ]:
dx = pd.read_csv('/content/REGISTROS_ACTIVIDAD_CONCESIONARIOS__ESTABLECIMIENTOS_VENTA_DE_AUTOMOVILES__2017_-_2021__4_AGO_2021_CAMARA_DE_COMERCIO_DE_VILLAVICENCIO_.csv')

In [ ]:
dx.shape

(613, 22)

In [ ]:
dx.dtypes

MATRICULA             object
ORGANIZACION          object
CATEGORIA             object
EST-MATRICULA         object
RAZON SOCIAL          object
FEC-MATRICULA          int64
FEC-RENOVACION         int64
ULT-ANO_REN            int64
DIR-COMERCIAL         object
MUN-COMERCIAL         object
TEL-COM               object
EMAIL-COMERCIAL       object
DIR-NOTIFICACION      object
MUN-NOTIFICACION      object
TEL-NOTF              object
EMAIL-NOTIFICACION    object
CIIU                  object
MAT-PROP.             object
CAM-PROP.             object
NOM-PROP              object
NOM-REP-LEGAL         object
NOMBRE-SUPLENTE       object
dtype: object

In [ ]:
dx.head(250)

MATRICULA                        ORGANIZACION  \
0      156750         Establecimiento de comercio   
1      296260         Establecimiento de comercio   
2      218975         Establecimiento de comercio   
3      319641         Establecimiento de comercio   
4      319592         Establecimiento de comercio   
..        ...                                 ...   
245    377860  Sociedad por acciones simplificada   
246    394596                     Persona natural   
247    364751                     Persona natural   
248    245559  Sociedad por acciones simplificada   
249    359964  Sociedad por acciones simplificada   

                                       CATEGORIA       EST-MATRICULA  \
0                                        Agencia    Matricula Activa   
1    No aplica el campo para el tipo de registro    Matricula Activa   
2    No aplica el campo para el tipo de registro    Matricula Activa   
3    No aplica el campo para el tipo de registro  Matricula Inactiva   
4    No aplica el campo para el tipo de registro    Matricula Activa   
..                                           ...                 ...   
245                                    Principal    Matricula Activa   
246  No aplica el campo para el tipo de registro    Matricula Activa   
247  No aplica el campo para el tipo de registro    Matricula Activa   
248                                    Principal    Matricula Activa   
249                                    Principal    Matricula Activa   

                                          RAZON SOCIAL  FEC-MATRICULA  \
0                                             MOTORYSA       20070524   
1                             COMPRA Y VENTA PORCELANA       20160526   
2    AUTOMOTORES COMERCIALES AUTOCOM S.A. VILLAVICE...       20110728   
3                   COMPRA Y VENTA VEHICULOS LLANO CAR       20170823   
4                        CONSIGNATARIA AUTOS MARANATHA       20170823   
..                                                 ...            ...   
245                               CODINVERSIONES S.A.S       20200917   
246                  BENAVIDES PEÑARANDA JOHAN SNEIDER       20210519   
247                      ORTIZ JIMENEZ ANDRES MAURICIO       20200129   
248                                     BIOINGEMET SAS       20130226   
249                         INVERSIONES NAMALUDA S.A.S       20191019   

     FEC-RENOVACION  ULT-ANO_REN                             DIR-COMERCIAL  \
0          20210325         2021                          CALLE 1 N 37-200   
1          20200128         2020                        CALLE 39C N. 7A-02   
2          20210330         2021               KR 22 3A 82 AV PUERTO LOPEZ   
3          20170823         2017                          AV 40 NRO. 25-35   
4          20190401         2019                          AV 40 NRO. 25-35   
..              ...          ...                                       ...   
245        20210514         2021               CL 35  13A  04 BRR CASTILLA   
246        20210519         2021                CL 35 20 C 13 BRR VAINILLA   
247        20200129         2020  CL 35 C 9 A 01 ESTE BRR VILLAS DE CATAMA   
248        20210401         2021                               CL 35 40 89   
249        20210330         2021                  CL 36 33 A 24 BRR BARZAL   

             MUN-COMERCIAL  ...  \
0    50001 - VILLAVICENCIO  ...   
1    50001 - VILLAVICENCIO  ...   
2    50001 - VILLAVICENCIO  ...   
3    50001 - VILLAVICENCIO  ...   
4    50001 - VILLAVICENCIO  ...   
..                     ...  ...   
245  50001 - VILLAVICENCIO  ...   
246  50001 - VILLAVICENCIO  ...   
247  50001 - VILLAVICENCIO  ...   
248  50001 - VILLAVICENCIO  ...   
249  50001 - VILLAVICENCIO  ...   

                                      DIR-NOTIFICACION       MUN-NOTIFICACION  \
0                                     AK 68 NO. 68B-61         11001 - BOGOTA   
1                                 ANILLO VIAL CRA 1-25  50001 - VILLAVICENCIO   
2    AUTOPISTA MEDELLIN 

Juan Pablo
1.¿Que tipo de organizacion cuenta con mas matriculas activas?

In [ ]:
razonSocial = dx[['ORGANIZACION', 'EST-MATRICULA']]
razonSocial.shape

(613, 2)

In [ ]:
alt.Chart(razonSocial).mark_bar().encode(    # Making a bar chart
    alt.X('count():Q'),                    # X-axis is an aggregated value for categorical variable in Y
    alt.Y('ORGANIZACION:N', sort = '-x'),   # Y-axis is sorted based on aggregated value
    alt.Color('EST-MATRICULA:N'),
    tooltip = [alt.Tooltip('count():Q')]
).properties(
    width = 600,
    height = 150
)


alt.Chart(...)

# Nueva sección

En la anterior grafica se puede filtrar la información de que las la mayoria de personas naturales cuentan con una matricula activa siendo menor la cantidad de matriculas inactivas




2. Ultimo año renovado con el tipo de organizacion

In [ ]:
lastRe = dx[['ORGANIZACION', 'ULT-ANO_REN']]
lastRe.shape

(613, 2)

In [ ]:
alt.Chart(lastRe).mark_bar().encode(    # Making a bar chart
    alt.X('count():Q'),                    # X-axis is an aggregated value for categorical variable in Y
    alt.Y('ORGANIZACION:N', sort = '-x'),   # Y-axis is sorted based on aggregated value
    alt.Color('ULT-ANO_REN:N'),
    tooltip = [alt.Tooltip('count():Q')]
).properties(
    width = 600,
    height = 150
)


alt.Chart(...)

En la anterior grafica se puede filtrar la información de la mayoria de organizaciones cuentan con su renovacion al dia, es decir la cantidad de empresas que deben actualizar son muy pocas comparado con las empresas ya totalmente renovadas

In [ ]:
tipoDeEmpresa = dx[['CATEGORIA','ORGANIZACION']]
tipoDeEmpresa.shape

(613, 2)

In [ ]:
alt.Chart(tipoDeEmpresa).mark_point(filled = True).encode(
    alt.X('ORGANIZACION'),
    alt.Y('CATEGORIA'),
    alt.Size('count():Q'),
    alt.Color('ORGANIZACION'),    # Using the color channel for encoding a categorical variable
    alt.OpacityValue(3)     # Adding size as channel for encoding a third quantitative variable
).properties(
    width = 600,
    height = 500
)

alt.Chart(...)

Que cantidad de categorias hay por tipo de organizacion?
Se pudo dilucidar de la informacion recolectada que se encuentra un tipo de empresa conocido como establemciento de comercio que no tienen nigun tipo de categoria, que no aplican porque suelen ser tipos de tiendas normales, no son una agencia de una empresa mas grande, son empresas que tienen una sola sede, se decidio utilizar este tipo de diagrama por su vista mas didactica frente a la cantidad de datos ya que tenemos referencias de que cantidad hay en base al tamaño del punto

In [ ]:
municipo = dx[['MUN-NOTIFICACION', 'MUN-COMERCIAL']]

In [ ]:
alt.Chart(municipo)\
    .mark_point().encode(            # Defining point as mark
        alt.X('count()'),  # Definig position in X as channel for encoding a quantitative variable
        alt.Y('MUN-NOTIFICACION'),
        alt.Size('count()'),
    alt.Color('MUN-NOTIFICACION'),    # Using the color channel for encoding a categorical variable
    alt.OpacityValue(2)     # Definig position in Y as channel for encoding other quantitative variable
    ).properties(
        width = 600,
        height = 500
    )

alt.Chart(...)

¿Cuantas empresas estan registradas fuera de villavicencio?
Se puede dar cuenta que aproximadamente el 39% de empresas no tienen sede principal en villavicencio, solo tienen una sede y/o agencia en villavicencio, se utilizo este diagrama de puntos donde se le asigno a colores diferentes a cada municipio para poder diferenciar de manera correcta cada uno.